### ISSUE TITLE
AllowSwappingAvatars world permission could disable loading cloud avatars

### ISSUE THREAD

**Issue by shiftyscales** [Frooxius/NeosPublic#669]
The "allow swapping avatars" world permission could prevent loading cloud avatars.

It allows session hosts the option to prevent potential attack vectors, e.g. Logix/particle bombs, etc. that have been affecting our educators because they keep their sessions open.

Because the cloud avatars are loaded- it's also possible to circumvent the AllowSwappingAvatars permission by favoriting a new avatar, and respawning, breaking its intended use.

It wouldn't need to be set as the default permission for spectators, so it'd need to be enabled per-session/per-world.

**Comment by ProbablePrime** [comment]
I disagree, loading cloud avatars can be blocked via the User Spawner controls. This component is old and needs to be updated to feature the new permission groups though. 

If the user spawner component would have the new permission groups this setting need not be overloaded.

**Comment by shiftyscales** [comment]
That granularity was my main concern, yeah. Being able to disable loading of cloud avatars per role.

If the CommonAvatarBuilder were updated to be in-line with the permission system, that would work too.

On that note- there are other aspects of the CommonAvatarBuilder I believe are out of date/deprecated. E.g. the "SetupCircularMenus" option is in reference to the old hand menus, which no longer exist.

**Comment by ProbablePrime** [comment]
Yup, that whole component can do a ton of powerful things but its slowly breaking with new systems.

**Comment by Hayden-da-Serg** [comment]
Yes, me and Prime saw that it doesn't apply for the whole permission set, and I also discussed this with Veer in Neos, and he also agrees that the entire permission set should be exposed in what avatar is applied to where in the CommonAvatarBuilder instead of just Host, Guests Only, or Anyone.

**Comment by ProbablePrime** [comment]
Just for quick reference, here's how you block cloud avatars currently(without any changes in this issue): https://youtu.be/eU8QHFiWDLY
